app/jobs/materialize_permissions.rb

```diff
@@ -0,0 +1,68 @@
+# A job the update a materialized of which users
+# have access to which clients.
+# At this point all it does is walk the existing scopes
+# and output timing data
+
+require 'ruby-progressbar'
+require 'pp'
+require 'benchmark'
+
+class MaterializePermissions < BaseJob
+  COLS = ['user_id', 'client_id', 'viewable'].freeze
+
+  def perform
+    # WIP:
+    # TODO: figure out how to handle other Client scopes...
+    # TODO: what about (soft) deletes of both users and clients
```

Comment:
Users and clients are soft-deleted, though clients may eventually disappear completely.